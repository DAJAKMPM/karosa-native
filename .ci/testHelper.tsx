import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
