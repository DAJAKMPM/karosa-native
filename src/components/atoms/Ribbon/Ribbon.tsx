/**
 *
 * Ribbon
 * @format
 *
 */

import React, { FC } from "react";
import { View } from "react-native";
import Text from "@app/atoms/Text";

import type { PropsType } from "./types";
import RibbonStyles from "./styles";

const Ribbon: FC<PropsType> = (props) => {
  const { upperText, lowerText } = props;

  return (
    <View style={RibbonStyles.baseContainer}>
      <View style={RibbonStyles.textContainer}>
        <Text customStyle={RibbonStyles.text} text={upperText} />
        <Text customStyle={RibbonStyles.text} text={lowerText} />
      </View>
      <View style={RibbonStyles.invertTriangleLeft} />
      <View style={RibbonStyles.invertTriangleRight} />
    </View>
  );
};

export default Ribbon;
