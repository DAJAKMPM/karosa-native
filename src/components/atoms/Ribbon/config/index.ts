/**
 *
 * @format
 *
 */

export const SIZE = {
  HEIGHT: 37,
  WIDTH: 30,
};
