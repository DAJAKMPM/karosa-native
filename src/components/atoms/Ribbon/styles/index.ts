/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme, DIMENS, SPACING } from "@app/styles";

const RibbonStyles = StyleSheet.create({
  baseContainer: {
    zIndex: 1,
  },
  textContainer: {
    width: DIMENS.screenWidth * 0.09,
    paddingTop: SPACING.xss,
    height: 37,
    backgroundColor: theme.colors.orange10,
    zIndex: 1,
  },
  text: {
    ...theme.textLightBold,
    fontWeight: "700",
    color: theme.colors.white,
    textAlign: "center",
    marginTop: 2,
  },
  invertTriangleLeft: {
    width: 0,
    height: 0,
    borderStyle: "solid",
    borderLeftWidth: DIMENS.screenWidth * 0.03,
    borderRightWidth: DIMENS.screenWidth * 0.029,
    borderBottomWidth: DIMENS.screenWidth * 0.06,
    backgroundColor: theme.colors.transparent,
    borderLeftColor: theme.colors.transparent,
    borderRightColor: theme.colors.transparent,
    borderBottomColor: theme.colors.orange10,
    transform: [{ rotate: "90deg" }],
    position: "absolute",
    top: 24,
  },
  invertTriangleRight: {
    width: 0,
    height: 0,
    borderStyle: "solid",
    borderLeftWidth: DIMENS.screenWidth * 0.029,
    borderRightWidth: DIMENS.screenWidth * 0.031,
    borderBottomWidth: DIMENS.screenWidth * 0.06,
    backgroundColor: theme.colors.transparent,
    borderLeftColor: theme.colors.transparent,
    borderRightColor: theme.colors.transparent,
    borderBottomColor: theme.colors.orange10,
    transform: [{ rotate: "-90deg" }],
    position: "absolute",
    top: 24,
    right: 0,
  },
});

export default RibbonStyles;
