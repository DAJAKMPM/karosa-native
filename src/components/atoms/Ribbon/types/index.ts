/**
 *
 * @format
 *
 */

export type PropsType = {
  upperText: string;
  lowerText: string;
};
