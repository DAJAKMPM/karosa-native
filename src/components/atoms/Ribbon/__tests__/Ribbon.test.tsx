import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Ribbon from "../Ribbon";

describe("Ribbon Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<Ribbon />);

    expect(toJSON()).toMatchSnapshot();
  });
});
