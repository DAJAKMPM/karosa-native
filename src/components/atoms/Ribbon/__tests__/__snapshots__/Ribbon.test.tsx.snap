// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribbon Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View>
    <View>
      <SvgMock
        height={37}
        width={30}
      />
    </View>
    <View
      style={
        Object {
          "bottom": 2,
          "position": "absolute",
          "right": -2,
          "textAlign": "center",
          "top": 2,
          "width": 67.5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "SFProText-Bold",
            "fontSize": 12,
            "fontWeight": "700",
            "lineHeight": 14.06,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "SFProText-Bold",
            "fontSize": 12,
            "fontWeight": "700",
            "lineHeight": 14.06,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
